### PR TITLE
Add docs for user roles

### PR DESCRIPTION
**Pull request by eleftherias**
# Summary

## Change Type

***Mark the type of change your PR introduces:***

- [ ] Bug fix (resolves an issue without affecting existing features)
- [ ] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [x] Documentation (updates or additions to documentation)
- [ ] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

***Outline how the changes were tested, including steps to reproduce and any relevant configurations. 
Attach screenshots if helpful.***

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [ ] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [ ] Included tests that validate the fix or feature.
- [ ] Checked that related changes are merged.

### CODE REVIEW

File: docs/docs/user_management/user_roles.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
 # User Roles in Minder
 
-Explains user roles
+When incorporating a user into your project, it's crucial to assign them the appropriate role based on their responsibilities and required access levels.
+Minder currently offers the following roles:
+
+- `viewer`: Provides read-only access to the project. Users with this role can view associated resources such as enrolled repositories, rule types, and profiles.
```
Comment (evankanderson) [inline docs/docs/user_management/user_roles.md:6]: I think they can also see the status of rule evaluations, right?

File: docs/docs/user_management/user_roles.md [hunk 2]
```diff
@@ -1,3 +1,10 @@
...
+- `editor`: Grants the same permissions as the viewer role, along with the ability to edit project resources, excluding the project itself and the list of providers.
```
Comment (evankanderson) [inline docs/docs/user_management/user_roles.md:7]: How about "in addition to the viewer permissions, editors can author profiles and rule types, as well as add resources to manage.  Editors _cannot_ enroll additional providers or change or delete projects."

File: docs/docs/user_management/user_roles.md [hunk 3]
```diff
@@ -1,3 +1,10 @@
...
+- `admin`: Grants administrator rights on the project. Users with this role have the same permissions as editor and can also modify the project and associated providers.
```
Comment (evankanderson) [inline docs/docs/user_management/user_roles.md:8]: Do you want to say "full permissions" here?  Maybe also mention that administrators can add and remove users from other roles?

File: docs/docs/user_management/user_roles.md [hunk 4]
```diff
@@ -1,3 +1,10 @@
...
+- `policy_writer`: Allows users to create rule types and profiles.
```
Comment (evankanderson) [inline docs/docs/user_management/user_roles.md:9]: Is this also granted by editor?  Might be worth contrasting the two.

**Review by evankanderson — DISMISSED** [review]

**Review by dmjb — APPROVED** [review]

**Review by ethomson — APPROVED** [review]

**Pull request merged**